Quorum v2.1.0

marcosio opened the pull request:
Updating to quorum v2.1.0

esanchma commented:
On hold until we can test this. @marcossanlab should we hardfork Alastria if we deploy this? Ideally, we would gradually update Quorum

marcosio commented:
If we integrate when they publish a new version, we are hand on with JP Morgan's Quorum and with the time, we will not need to maintain a hard/light fork.

If we can use two different versions of Quorum in our network, of course, we would update. On the other hand, if it will not be possible, we need to study the better way to upgrade the network.

marcosio commented:
Please, review the rebase on brach: 

https://github.com/alastria/quorum/tree/Quorum_2_1_0_Rebase

fjl committed:
core/state: simplify proof methods (#17965)

This fixes the import cycle build error in core/vm tests.
There is no need to refer to core/vm for a type definition.

jsvisa committed:
eth/downloader: SetBlocksIdle is not used (#17962)

   __
 <(o )___
  ( ._> /
   `---'

Commit:
Revert "Fix vendor package error in Mojave"

This reverts commit e679587e1f0059ff931d8a88f373b5336d25037a.

Commit:
merge with upstream/master

SatpalSandhu61 committed:
Remove logging used for investigation.

jbhurat committed:
Adding private log blooms to DB in addition to public log bloom

johnsBeharry committed:
cmd/clef: replace password arg with prompt (#17897)

* cmd/clef: replace password arg with prompt (#17829)

Entering passwords on the command line is not secure as it is easy to recover from bash_history or the process table.
1. The clef command addpw was renamed to setpw to better describe the functionality
2. The <password> argument was removed and replaced with an interactive prompt

* cmd/clef: remove undeclared variable

holisticode committed:
p2p accounting (#17951)

* p2p/protocols: introduced protocol accounting

* p2p/protocols: added TestExchange simulation

* p2p/protocols: add accounting simulation

* p2p/protocols: remove unnecessary tests

* p2p/protocols: comments for accounting simulation

* p2p/protocols: addressed PR comments

* p2p/protocols: finalized accounting implementation

* p2p/protocols: removed unused code

* p2p/protocols: addressed @nonsense PR comments

holiman committed:
core/vm: check empty in extcodehash

frncmx committed:
swarm: clean up unused private types and functions (#17989)

* swarm: clean up unused private types and functions

Those that were identified by code inspection tool.

* swarm/storage: move/add Proximity GoDoc from deleted private function

The mentioned proximity() private function was deleted in:
1ca8fc1e6fa0ab4ab1aaca06d6fb32e173cd5f2f

rociiu committed:
cmd/swarm/swarm-smoke: fix issue that loop variable capture in func (#17992)

acud committed:
all: updated code owners file (#17987)

amalrajmani committed:
Merge pull request #7 from SatpalSandhu61/geth-upgrade-1.8.12-fix-for-large-gas-estimate

Geth upgrade 1.8.12 fix for large gas estimate

SatpalSandhu61 committed:
Merge branch 'master' of https://github.com/jpmorganchase/quorum

acud committed:
cmd/swarm: auto resolve default path according to env flag (#17960)

KimMachineGun committed:
swarm: modify context key (#17925)

* swarm: modify context key

* gofmt sctx.go

nonsense committed:
swarm/api: unexport Respond methods (#18037)

holisticode committed:
swarm: Better syncing and retrieval option definition (#17986)

* swarm: Better syncing and retrieval option definition

* swarm/network/stream: better comments

* swarm/network/stream: addressed PR comments

rjl493456442 committed:
miner: fix miner stress test (#18039)

SamuelMarks committed:
travis, appveyor: bump to Go 1.11.2 (#18031)

CoreyLin committed:
p2p/protocols: use keyed fields for struct instantiation (#18017)

holiman committed:
downloader: measure successfull deliveries, not failed (#17983)

* downloader: measure successfull deliveries, not failed

* downloader: fix typos

holiman committed:
core, eth/downloader: fix validation flaw, fix downloader printout flaw (#17974)

holiman committed:
signer: remove ineffectual assignments (#18049)

* signer: remove ineffectual assignments

* signer: remove ineffectual assignments

jsvisa committed:
eth/downloader: remove the expired id directly (#17963)

jpeletier committed:
swarm/api: Fix #18007, missing signature should return HTTP 400 (#18008)

SatpalSandhu61 committed:
Revert fix for eth_estimateGas due to side-effect where attempt to purely read private state doesn't work correctly.

fjl committed:
eth/downloader: speed up tests by generating chain only once (#17916)

* core: speed up GenerateChain

Use a mock implementation of ChainReader instead of creating
and destroying a BlockChain object for each generated block.

* eth/downloader: speed up tests by generating chain only once

This change reworks the downloader tests so they share a common test
blockchain instead of generating a chain in every test. The tests are
roughly twice as fast now.

jpmsam committed:
Merge pull request #556 from jbhurat/missing-private-bloom-state

Writing private block bloom for miner

jpmsam committed:
Merge pull request #555 from vdamle/permissioned-log

p2p: remove spurious log for denied node permissioning

Puneetha17 committed:
Do not reset V value if the tx is already set to private

vmaark committed:
swarm/network: light nodes are not dialed, saved and requested from (#17975)

* RequestFromPeers does not use peers marked as lightnode

* fix warning about variable name

* write tests for RequestFromPeers

* lightnodes should be omitted from the addressbook

* resolve pr comments regarding logging, formatting and comments

* resolve pr comments regarding comments and added a missing newline

* add assertions to check peers in live connections

nonsense committed:
swarm, cmd/swarm: address ineffectual assignments (#18048)

* swarm, cmd/swarm: address ineffectual assignments

* swarm/network: remove unused vars from testHandshake

* swarm/storage/feed: revert cursor changes

amalrajmani committed:
Merge pull request #8 from SatpalSandhu61/geth-upgrade-1.8.12-revert-estimategas-fix

Revert fix for eth_estimateGas due to side-effect where attempt to pu…

kurkomisi committed:
p2p: use enode.ID type in metered connection (#17933)

Change the type of the metered connection's id field from string to enode.ID.

CoreyLin committed:
event, event/filter: minor code cleanup (#18061)

LiangMa committed:
p2p: fix comment typo (#18027)

CoreyLin committed:
eth, p2p: fix comment typos (#18014)

CoreyLin committed:
core/vm, eth/tracers: use pointer receiver for GetRefund (#18018)

JoranHonig committed:
common/compiler: capture runtime code and source maps (#18020)

ryanschneider committed:
internal/web3ext: add eth.getProof (#18052)

karalabe committed:
Merge pull request #17982 from holiman/polish_contantinople_extcodehash

core/vm: check empty in extcodehash

fjl committed:
event/filter: delete unused package (#18063)

Commit:
ethclient: include block hash from FilterQuery (#17996)

ethereum/go-ethereum#16734 introduced BlockHash to the FilterQuery
struct. However, ethclient was not updated to include BlockHash in the actual
RPC request.

rjl493456442 committed:
consensus: extend getWork API with block number (#18038)

fjl committed:
core/state: remove lock (#18065)

The lock in StateDB is useless. It's only held in Copy, but Copy is safe
for concurrent use because all it does is read.

kurkomisi committed:
metrics, p2p: add ephemeral registry (#18067)

* metrics, p2p: add ephemeral registry

* metrics: fix linter issue

CoreyLin committed:
rawdb: remove unused parameter for WritePreimages func (#18059)

* rawdb: remove unused parameter for WritePreimages func and modify a
spelling mistake

* rawdb: update the doc for function WritePreimages

randomshinichi committed:
Imply that SwarmApiFlag is the API endpoint to connect to, not to listen on (#18071)

nolash committed:
p2p/simulations, swarm/network: Custom services in snapshot (#17991)

* p2p/simulations: Add custom services to simnodes + remove sim down conn objs

* p2p/simulation, swarm/network: Add selective services to discovery sim

* p2p/simulations, swarm/network: Remove useless comments

* p2p/simulations, swarm/network: Clean up mess from rebase

* p2p/simulation: Add sleep to prevent connect flakiness in http test

* p2p/simulations: added concurrent goroutines to prevent sleeps on simulation connect/disconnect

* p2p/simulations, swarm/network/simulations: address pr comments

* reinstated dummy service

* fixed http snapshot test

rjl493456442 committed:
core: fix default trie cache limit (#17860)

frncmx committed:
swarm/storage: fix access count on dbstore after cache hit (#17978)

Access count was not incremented when chunk was retrieved
from cache. So the garbage collector might have deleted the most
frequently accessed chunk from disk.

Co-authored-by: Ferenc Szabo <ferenc.szabo@ethereum.org>

mrFranklin committed:
github: format code owners file (#18090)

replace tabs by spaces in the code owners file

mrFranklin committed:
eth: fix minor grammar issue in comment (#18091)

frncmx committed:
swarm/storage: fix garbage collector index skew (#18080)

On file access LDBStore's tryAccessIdx() function created a faulty
GC Index Data entry, because not indexing the ikey correctly.
That caused the chunk addresses/hashes to start with '00' and the last
two digits were dropped. => Incorrect chunk address.

Besides the fix, the commit also contains a schema change which will
run the CleanGCIndex() function to clean the GC index from erroneous
entries.

Note: CleanGCIndex() rebuilds the index from scratch which can take
a really-really long time with a huge DB (possibly an hour).

karalabe committed:
params: update CHTs

karalabe committed:
Merge pull request #18097 from karalabe/update-chts-2

params: update CHTs

nizsheanez committed:
cmd/swarm, swarm/api/http, swarm/bmt, swarm/fuse, swarm/network/stream, swarm/storage, swarm/storage/encryption, swarm/testutil: use pseudo-random instead of crypto-random for test files content generation (#18083)

- Replace "crypto/rand" to "math/rand" for files content generation
- Remove swarm/network_test.go.Shuffle and swarm/btm/btm_test.go.Shuffle - because go1.9 support dropped (see https://github.com/ethereum/go-ethereum/pull/17807 and comments to swarm/network_test.go.Shuffle)

karalabe committed:
params, swarm: release Geth v1.8.18 and Swarm v0.3.6

michaelgu95 committed:
Add --txsizelimit flag to geth CLI that allows expansion of transaction size limit used by tx_pool.go => validateTx()

Commit:
update gas limit as per quorum upstream

Commit:
update gas limit as per quorum upstream

Commit:
revert to getDualState instead of evm.StateDB as per quorum

Commit:
core/vm : remove get dual state calls in some op codes as its unnecessary. evm push is setting stateDB to public / private correctly.

SatpalSandhu61 committed:
Merge branch 'master' of https://github.com/jpmorganchase/quorum

jpmsam committed:
Merge pull request #510 from nathanawmk/master

Update max transaction size limit.

trung committed:
fix unit test for PR 510

jpmsam committed:
Merge pull request #577 from trung/fix-unit-test

fix unit test for PR 510

Commit:
Merge remote-tracking branch 'quorum/master' into geth-upgrade-1.8.12

# Conflicts:
#	core/tx_pool_test.go
#	miner/worker.go

Commit:
Merge remote-tracking branch 'quorum/master' into geth-upgrade-1.8.12

# Conflicts:
#	core/tx_pool_test.go
#	miner/worker.go

Commit:
core/vm: revert to joel's change to getdualstate. This is a mandatory change for quorum

dual state & read only EVM

    This commit implements a dual state approach. The dual state approach
    separates public and private state by making the core vm environment
    context aware.

    Although not currently implemented it will need to prohibit value
    transfers and it must initialise all transactions from accounts on the
    public state. This means that sending transactions increments the
    account nonce on the public state and contract addresses are derived
    from the public state when initialised by a transaction. For obvious
    reasons, contract created by private contracts are still derived from
    public state.

    This is required in order to have consensus over the public state at all
    times as non-private participants would still process the transaction on
    the public state even though private payload can not be decrypted. This
    means that participants of a private group must do the same in order to
    have public consensus. However the creation of the contract and
    interaction still occurs on the private state.

    It implements support for the following calling model:

    S: sender, (X): private, X: public, ->: direction, [ ]: read only mode

    1. S -> A -> B
    2. S -> (A) -> (B)
    3. S -> (A) -> [ B -> C ]

    It does not support

    1. (S) -> A
    2. (S) -> (A)
    3. S -> (A) -> B

    Implemented "read only" mode for the EVM. Read only mode is checked
    during any opcode that could potentially modify the state. If such an
    opcode is encountered during "read only", it throws an exception.

    The EVM is flagged "read only" when a private contract calls in to
    public state.

vsmk98 committed:
changes to set readOnlyDepth only once during the entire opCode execution

fixanoid committed:
Update HACKING.md

jpmsam committed:
Merge pull request #542 from vsmk98/geth-upgrade-1.8.12

Upgrade Quorum to geth 1.8.12

trung committed:
fixed method signature

jpmsam committed:
Merge pull request #581 from trung/fix-tx_pool_test

Fix method signature for txpool test.

Commit:
Merge remote-tracking branch 'quorum/master' into geth-upgrade-1.8.16

Commit:
Merge remote-tracking branch 'quorum/master' into geth-upgrade-1.8.16

SatpalSandhu61 committed:
Merge branch 'master' of https://github.com/jpmorganchase/quorum

jbhurat committed:
Setting the private state to public state if contract address is not present in the private state

trung committed:
tests/: minor refactoring to enable Ethereum tests

trung committed:
simplify the setup

jpmsam committed:
Merge pull request #570 from jbhurat/missing-logs-events

Adding private log blooms to DB in addition to public log bloom

fixanoid committed:
Rename HACKING.md to NOTES.md

fixanoid committed:
Update NOTES.md

trung committed:
separated consts for Quorum and those used in upstream Geth

fixanoid committed:
Moving Raft doc into docs

fixanoid committed:
Updating raft doc location

fixanoid committed:
Adding istanbul-tools ref

fixanoid committed:
Istanbul RPC API

fixanoid committed:
Update README.md

fixanoid committed:
Updating IBFT API doc location

jpmsam committed:
Merge pull request #586 from trung/fix-ethereum-tests

tests/: minor refactoring to enable Ethereum tests

fixanoid committed:
Update BUILDING.md

namtruong committed:
add support for private sendRawTransaction - merge with latest quorum

trung committed:
added dual state implementation to StaticCall

michaelgu95 committed:
Merge branch 'master' into photic-tx-size-limit

SatpalSandhu61 committed:
Fix for issue 576 where private transactions can cause block gas limit to be exceeded, leading to BAD BLOCK.

SatpalSandhu61 committed:
Fix for issue 576 where private transactions can cause block gas limit to be exceeded, leading to BAD BLOCK.

trung committed:
merged from upstream/master

trung committed:
fixed unit test

jpmsam committed:
Merge pull request #585 from jbhurat/estimate-gas

Fixing high estimate gas issue

trung committed:
staticcall does not modify states so no need to push to stack

jpmsam committed:
quorum version update

SatpalSandhu61 committed:
Merge branch 'master' of https://github.com/jpmorganchase/quorum

fixanoid committed:
Cherrypicking go-ethereum#18211 pull

jbhurat committed:
calling unlock using defer, preventing deadlock when miner.setEtherbase() returns an error

jpmsam committed:
Merge pull request #598 from jbhurat/miner-setcoinbase

Prevent deadlock when miner.setEtherbase is called for Istanbul consensus

jpmsam committed:
Merge pull request #597 from jpmorganchase/fd-limit-max-out

Maxing out fdlimit per go-ethereum#18211

jpmsam committed:
Merge pull request #592 from trung/staticcall-dual-state

added dual state implementation to StaticCall

prd-fox committed:
Allow processing of sync content whilst it is being downloaded when
using a bounded sync

namtruong committed:
return error when sendRawPrivateTransaction is called to submit public transaction (privateFor list empty)

namtruong committed:
minor documentation update

jpmsam committed:
Merge pull request #591 from QuorumEngineering/feature/support-private-sendRawTransaction-merge

feature/support private sendRawTransaction

prd-fox committed:
Added comment to explain the change

jpmsam committed:
Merge pull request #563 from Puneetha17/private_rawtx

Do not reset V value if the tx is already set to private

fixanoid committed:
Adding Quorum info into nodeInfo

fixanoid committed:
Updating test

SatpalSandhu61 committed:
Merge branch 'master' of https://github.com/jpmorganchase/quorum

SatpalSandhu61 committed:
Currently, we can't tell the reason for evm failure, so change log message for EVM failure reason from Debug to Info. This is helpful to users and makes analysis of issues much easier.

Commit:
fixgo11

fixanoid committed:
Quorum version bump

trung committed:
Minor update to README (#622)

* Added full logo link
* Added download link

trung committed:
Binaries distribution to Bintray (#623)

distribute binaries to bintray when tagging

jpmsam committed:
Merge pull request #618 from SatpalSandhu61/print-vm-failure-reason-to-log

Print vm failure reason to log

Commit:
Merge remote-tracking branch 'remotes/origin/master' into geth-upgrade-1.8.18

# Conflicts:
#	consensus/ethash/consensus.go
#	eth/bloombits.go
#	params/protocol_params.go

Commit:
Merge remote-tracking branch 'remotes/origin/master' into geth-upgrade-1.8.18

# Conflicts:
#	consensus/ethash/consensus.go
#	eth/bloombits.go
#	params/protocol_params.go

Commit:
Merge branch 'goeth-1.8.18' into geth-upgrade-1.8.18

# Conflicts:
#	.github/CONTRIBUTING.md
#	.travis.yml
#	core/blockchain.go
#	core/vm/evm.go
#	core/vm/logger_test.go
#	eth/handler_test.go
#	eth/tracers/tracer_test.go
#	internal/ethapi/api.go
#	internal/web3ext/web3ext.go
#	p2p/discover/node.go
#	p2p/server.go
#	swarm/network/simulations/discovery/discovery_test.go

Commit:
Merge branch 'goeth-1.8.18' into geth-upgrade-1.8.18

# Conflicts:
#	.github/CONTRIBUTING.md
#	.travis.yml
#	core/blockchain.go
#	core/vm/evm.go
#	core/vm/logger_test.go
#	eth/handler_test.go
#	eth/tracers/tracer_test.go
#	internal/ethapi/api.go
#	internal/web3ext/web3ext.go
#	p2p/discover/node.go
#	p2p/server.go
#	swarm/network/simulations/discovery/discovery_test.go

jpmsam committed:
Merge pull request #621 from QuorumEngineering/raft-sync-greater-height

Raft: process sync data in parallel to download

jpmsam committed:
Merge pull request #616 from jpmorganchase/quorum-node-info

Adding Quorum version into node info

jbhurat committed:
Merge branch 'master' into master

jpmsam committed:
Merge pull request #619 from drequinox/master

Fixes for Go 11

Commit:
merge 1.8.18 with quorum/master
refactor raft & istanbul based on upstream changes
fix unit test cases

Commit:
Merge remote-tracking branch 'remotes/origin/master' into geth-upgrade-1.8.18-latest

# Conflicts:
#	.travis.yml
#	cmd/geth/consolecmd_test.go
#	vendor/github.com/rjeczalik/notify/watcher_readdcw.go
#	vendor/golang.org/x/sys/cpu/cpu.go
#	vendor/golang.org/x/sys/cpu/cpu_arm.go
#	vendor/golang.org/x/sys/cpu/cpu_arm64.go
#	vendor/golang.org/x/sys/cpu/cpu_mips64x.go
#	vendor/golang.org/x/sys/cpu/cpu_mipsx.go
#	vendor/golang.org/x/sys/cpu/cpu_ppc64x.go
#	vendor/golang.org/x/sys/cpu/cpu_s390x.go
#	vendor/vendor.json

Commit:
fix merge conflict

Commit:
fix unit test

Commit:
update newly added methods

jpmsam committed:
Merge pull request #595 from SatpalSandhu61/fix-for-bad-block-on-gas-limit

Fix for BAD BLOCK caused by "gas limit reached" for private contracts

fixanoid committed:
Merge branch 'txn-size-limit-cli' into txn-size-limit-genesis

jbhurat committed:
Using NewBlockMsg constant instead of hardcoded 0x07

fixanoid committed:
Transaction size limit support via genesis

jpmsam committed:
Merge pull request #539 from jbhurat/impossible-reorg-fix-v1

consensus/istanbul: Race condition causing "Impossible reorg" error

Commit:
Merge remote-tracking branch 'remotes/origin/master' into AJ-geth-upgrade-1.8.18

fixanoid committed:
Renaming to TransactionSizeLimit for clarity

chris-j-h committed:
Raft doc and log message updates

prd-fox committed:
Compare intrinsic gas of public transaction versus the private version
that contains the PTM hash instead. If there is a difference, add it on
to the execution cost.

prd-fox committed:
Require the addition of the extra intrinsic gas cost to remain under the
user input gas limit.

amalrajmani committed:
travis: run quorum-acceptance-tests for pull requests (#637)

* update travis to enable running acceptance test for raft, istanbul and clique consensus when a PR is raised

Commit:
Merge remote-tracking branch 'remotes/origin/master' into AJ-geth-upgrade-1.8.18

Commit:
raft: change RAFTPORT to RaftPort

Commit:
remove TODO related to review

fixanoid committed:
Update BUILDING, adding Windows info

Commit:
update comments

Commit:
merge CONTRIBUTING.md with upstream

Commit:
update CONTRIBUTING.md with quorum wiki

Commit:
Merge remote-tracking branch 'remotes/origin/master' into AJ-geth-upgrade-1.8.18

jpmsam committed:
Merge pull request #647 from QuorumEngineering/raft-doc-update

Update Raft doc and log messages

trung committed:
use solc 0.5.4

trung committed:
grant executable to solc

jpmsam committed:
Merge pull request #648 from trung/solc-specific-version

Use solc 0.5.4 when running acceptance tests against pull requests

Commit:
Merge remote-tracking branch 'remotes/origin/master' into AJ-geth-upgrade-1.8.18

Commit:
solc upgrade

Commit:
remove comment

jpmsam committed:
Merge pull request #636 from jpmorganchase/txn-size-limit-genesis

Adding support for transaction size limit via genesis

jpmsam committed:
Merge pull request #644 from QuorumEngineering/estimategas-for-privatetransaction

EstimateGas to handle private transaction gas cost differences

jbhurat committed:
Adding a node only if it is not in the raft peer list

Commit:
merge with quorum upstream
params: config.go and denomination.go merged with conflicts

Commit:
merge with quorum upstream
params: config.go and denomination.go merged with conflicts

jpmsam committed:
Merge pull request #639 from QuorumEngineering/AJ-geth-upgrade-1.8.18

geth upgrade 1.8.18

jimthematrix committed:
Check for Sender() return errors before using the resolved sender address, to account for sending EIP155 externally signed txs before the EIP155 block

jimthematrix committed:
Added an Info log when skipping incorrectly signed txs; cleaned up debug logging

jpmsam committed:
Merge pull request #656 from jbhurat/raft-panic-adding-peer

Adding a node only if it is not in the raft peer list

chris-j-h committed:
Add encrypted tx payload hash from TM to Quorum tx's data field

The TM response was being stored in an isolated scope variable,
therefore the tx data was not being updated correctly for private txs.
This meant the private tx payload could not be retrieved from the TM
when needed.

jpmsam committed:
Merge pull request #667 from kaleido-io/raft-fix-master

Check recovered sender address before including txs in the TransactionsByPriceAndNonce list

jpmsam committed:
version update

jbhurat committed:
Reading default transaction size limit, if it is 0

jpmsam committed:
Merge pull request #675 from jbhurat/defaulting_tx_size_limit

Reading default transaction size limit, if it is 0

jpmsam committed:
version update

Commit:
raft: pass gas floor and gas ceil to calcGasLimit when minting block

Commit:
permission: set default gas ceil and floor to 700M which is the default minGasLimit

jbhurat committed:
Adding a lock around WriteBlockWithState while processing sealing result

Krish1979 committed:
Rename docs/README.md to docs/test/README.md

Krish1979 committed:
Rename docs/api.md to docs/test/api.md

Krish1979 committed:
Rename docs/design.md to docs/test/design.md

Krish1979 committed:
Rename docs/istanbul-rpc-api.md to docs/test/istanbul-rpc-api.md

Krish1979 committed:
Rename docs/privacy.md to docs/test/privacy.md

Krish1979 committed:
Rename docs/raft.md to docs/test/raft.md

Krish1979 committed:
Rename docs/running.md to docs/test/running.md

Krish1979 committed:
Rename docs/test/api.md to docs/api.md

Krish1979 committed:
Rename docs/test/README.md to docs/README.md

Krish1979 committed:
Rename docs/test/design.md to docs/design.md

Krish1979 committed:
Rename docs/test/istanbul-rpc-api.md to docs/istanbul-rpc-api.md

Krish1979 committed:
Rename docs/test/raft.md to docs/raft.md

Krish1979 committed:
Rename docs/test/privacy.md to docs/privacy.md

Krish1979 committed:
Rename docs/test/running.md to docs/running.md

jbhurat committed:
Checking if c.current is nil, as it can be when istanbul engine is stopped due to block download event

jbhurat committed:
Merge branch 'master' into lock_writewithblockstate

SatpalSandhu61 committed:
Update raft.md

Adding documentation that the timestamp is in nanoseconds.

jpmsam committed:
Merge pull request #697 from jpmorganchase/SatpalSandhu61-raft-doc-for-timestamp

Update raft.md

Krish1979 committed:
Quorum documentation update (#701)

Consolidate documentation for Quorum, its Privacy Manager and usage

namtruong committed:
Update documentation url (#702)

Krish1979 committed:
Update Setup Overview & Quickstart.md

Krish1979 committed:
Quorum documentation update (#703)

updated documentation to link to quorum examples repo.

fixanoid committed:
Logo update

Commit:
set the default ceil and floor gas limit to 800M and 700M

Commit:
set the default ceil and floor gas limit to 800M and 700M

Commit:
whisper test: reduce the number of nodes in TestSimulation to reduce memory consumption while running in travis
increase wait period for timeout

jbhurat committed:
Docs updates (#707)

chris-j-h committed:
Flesh out Tessera monitoring docs with more info and examples

Improve InfluxDB monitoring docs

Improve monitoring Grafana docs and include dashboard examples

Review Splunk monitoring docs

Expand Prometheus monitoring docs

Step by step instructions for creating a basic Grafana dashboard

Update Influx docs to include SSL config

jpmsam committed:
Merge pull request #716 from chris-j-h/tessera-monitoring-docs

Tessera monitoring documentation update

jbhurat committed:
Merge branch 'master' into lock_writewithblockstate

Krish1979 committed:
Update mkdocs.yml

Krish1979 committed:
Create FAQ.md

Krish1979 committed:
Update FAQ.md

namtruong committed:
update faq styling

Krish1979 committed:
Update FAQ.md

Krish1979 committed:
Update FAQ.md

namtruong committed:
update relative links

namtruong committed:
info admonition

jpmsam committed:
Merge pull request #684 from jbhurat/lock_writewithblockstate

Adding a lock around WriteBlockWithState while processing sealing result

jpmsam committed:
Merge branch 'master' into fix/go-test

trung committed:
encapsulated calculate gas limit logic

trung committed:
merged with upstream

trung committed:
cross reference MinerGasCeil

fixanoid committed:
Merge branch 'master' into master

jpmsam committed:
Merge pull request #717 from jpmorganchase/fix/go-test

whisper test: reduce the number of nodes in TestSimulation

trung committed:
Merge branch 'master' into fix/raft-calcgaslimit

Krish1979 committed:
Merge branch 'master' into master

trung committed:
ignore swarm package when running tests for PR

trung committed:
Merge branch 'master' into ignore-swarm-tests

trung committed:
updated quorum examples docs

trung committed:
Merge branch 'master' into update-docs

Krish1979 committed:
Merge pull request #720 from Krish1979/master

Updated FAQ Section in Quorum docs

Krish1979 committed:
Merge branch 'master' into update-docs

Commit:
Merge remote-tracking branch 'origin/fix/raft-calcgaslimit' into fix/raft-calcgaslimit-1

# Conflicts:
#	eth/config.go

Krish1979 committed:
Merge pull request #722 from trung/update-docs

updated quorum examples docs

trung committed:
Merge branch 'master' into ignore-swarm-tests

jpmsam committed:
Merge pull request #721 from trung/ignore-swarm-tests

ignore tests in `swarm` package for PR builds

jpmsam committed:
Merge branch 'master' into fix/raft-calcgaslimit

jpmsam committed:
Merge branch 'master' into bugfix/personal_sendTransaction

jpmsam committed:
Merge pull request #679 from jpmorganchase/bugfix/personal_sendTransaction

ETHAPI: Private tx bug fix for personal_sendTransaction API

jpmsam committed:
Merge branch 'master' into fix/raft-calcgaslimit

chris-j-h committed:
Add logo favicon and other minor doc fixes/changes: (#723)

- Fix incorrectly documented Tessera server config field name
- Add Quorum logo favicon
- Add 404.html from material theme to try fix 404 not displaying properly
- FAQ ordering and phrasing changes
- Fix formatting in Tessera TLS config docs

libby committed:
Adding signer tests for EIP155, Homestead and private tx. (#682)

Adding signer tests for EIP155, Homestead and private tx.

fixanoid committed:
Adding Quorum compatibility table to FAQ (#726)

Commit:
Merge remote-tracking branch 'origin/fix/raft-calcgaslimit' into fix/raft-calcgaslimit-1

Commit:
Merge remote-tracking branch 'remotes/origin/master' into fix/raft-calcgaslimit-1

jpmsam committed:
Merge pull request #680 from jpmorganchase/fix/raft-calcgaslimit

raft: pass gas floor and gas ceil to calcGasLimit when minting block

trung committed:
updated broken links (#728)
